### PR TITLE
[Documentation][Customization] Non-resource controllers - update

### PR DESCRIPTION
**Pull request by TheMadeleine**
| Q               | A
| --------------- | ---
| Doc fix?        | yes |
| New docs?    | no |
| Related tickets |  |
| License         | MIT |

### CODE REVIEW

File: docs/customization/controller.rst [hunk 1]
```diff
@@ -109,57 +110,42 @@ getting a list of recommended products from your external api.
...
-            $emailManager = $this->get('sylius.email_manager.order');
-            $emailManager->sendConfirmationEmail($order);
-
-            return $this->redirectToRoute('sylius_checkout_thank_you');
+            // put your custom logic here
         }
     }
 
-2. The next thing you have to do is to override the ``sylius.controller.frontend.order.class`` parameter in ``app/config/services.yml``.
+2. The next thing you have to do is to override the ``sylius.controller.shop.homepage.class`` parameter in the ``app/config/services.yml``.
```
Comment (pamil) [inline docs/customization/controller.rst]: We're not using this parameter anywhere, it won't work.

It is needed to override the whole `sylius.controller.shop.homepage` service definition.

File: docs/customization/controller.rst [hunk 2]
```diff
@@ -10,7 +10,8 @@ If you want to override a controller action, check which controller you should b
     There are two types of controllers we can define in Sylius:
 
     **Resource Controllers** - are based only on one Entity, so they return only the resources they have in their name. For instance a ``ProductController`` should return only products.
-    **Standard Controllers** - non-resource; these may use many entities at once, they are useful on more general pages. We are extending these controllers only if the actions we want cannot be done through yaml configuration - like sending emails.
+    **Standard Controllers** - non-resource; these may use many entities at once, they are useful on more general pages.
+    We are extending these controllers only if the actions we want cannot be done through yaml configuration - like sending emails.
```
Comment (lchrusciel) [inline docs/customization/controller.rst]: `extending` -> `defining`. We don't used to extend any class at all in standard controllers.

File: docs/customization/controller.rst [hunk 3]
```diff
@@ -109,57 +110,67 @@ getting a list of recommended products from your external api.
...
+Let's assume that you would like to add some logic to the Homepage.
 
-1. Create a new Controller class under the ``AppBundle/Controller/Frontend`` namespace.
+1. Create a new Controller class under the ``AppBundle/Controller/Shop`` namespace.
 
-Run ``$ php bin/console debug:container sylius.controller.frontend.order``.
+Run ``$ php bin/console debug:container sylius.controller.shop.homepage``.
 
-Your class needs to extend this base class.
+Your class needs to extend the Symfony's Controller, and if you still need the methods of the original HompageController,
```
Comment (lchrusciel) [inline docs/customization/controller.rst]: Can we write something like: `We encourage you to define the controllers as a service`. And also there is no need of extending Symfony's Controller

File: docs/customization/controller.rst [hunk 4]
```diff
@@ -109,57 +110,67 @@ getting a list of recommended products from your external api.
...
 
-    use Sylius\Bundle\WebBundle\Controller\Frontend\Account\OrderController as BaseOrderController;
-    use Sylius\Bundle\PayumBundle\Request\GetStatus;
+    use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+    use Symfony\Bundle\FrameworkBundle\Templating\EngineInterface;
     use Symfony\Component\HttpFoundation\Request;
     use Symfony\Component\HttpFoundation\Response;
 
-    class OrderController extends BaseOrderController
+    class HomepageController extends Controller
```
Comment (lchrusciel) [inline docs/customization/controller.rst]: final

File: docs/customization/controller.rst [hunk 5]
```diff
@@ -109,57 +110,67 @@ getting a list of recommended products from your external api.
...
 
 .. code-block:: yaml
 
-    parameters:
-        sylius.controller.frontend.order.class: AppBundle\Controller\Frontend\OrderController
+    services:
+        sylius.controller.shop.homepage: AppBundle\Controller\Shop\HomepageController
 
-From now on your ``afterPurchaseAction`` of the ``OrderController`` will also send emails in addition to its default behaviour.
+From now on your ``customAction`` of the ``HomepageController`` will be available alongside the ``indexAction`` from the base class.
```
Comment (lchrusciel) [inline docs/customization/controller.rst]: We should add some link to Symfony docs about controllers

**Comment by pjedrzejewski** [comment]
Thanks Magda!

**Pull request merged**